nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,178 @@
+            .name("delete-hb-flowfile-fetch-count")
+            .displayName("Flowfile Fetch Count")
+            .description("The number of flowfiles to fetch per run.")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("5")
+            .expressionLanguageSupported(false)
+            .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("delete-hb-row-ff-count")
+            .displayName("Batch Size")
+            .description("The number of deletes to send per batch.")
+            .required(true)
+            .defaultValue("50")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(false)
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = super.getSupportedPropertyDescriptors();
+        properties.add(ROW_ID_LOCATION);
+        properties.add(FLOWFILE_FETCH_COUNT);
+        properties.add(BATCH_SIZE);
+
+        return properties;
+    }
+
+    @Override
+    protected void doDelete(ProcessContext context, ProcessSession session) throws Exception {
+        final int batchSize     = context.getProperty(BATCH_SIZE).asInteger();
+        final String location   = context.getProperty(ROW_ID_LOCATION).getValue();
+        final int flowFileCount = context.getProperty(FLOWFILE_FETCH_COUNT).asInteger();
+        List<FlowFile> flowFiles = session.get(flowFileCount);
+
+        if (flowFiles != null && flowFiles.size() > 0) {
+            try {
+                if (location.equals(ROW_ID_BODY.getValue())) {
+                    doDeleteFromBody(flowFiles, context, session, batchSize);
+                } else {
+                    doDeleteFromAttributes(flowFiles, context, session, batchSize);
+                }
+
+                for (int index = 0; index < flowFiles.size(); index++) {
+                    session.transfer(flowFiles.get(index), REL_SUCCESS);
+                }
+            } catch (Exception ex) {
+                getLogger().error("");
+                for (int index = 0; index < flowFiles.size(); index++) {
+                    session.transfer(flowFiles.get(index), REL_FAILURE);
+                }
+            }
+        }
+    }
+
+    private void doDeleteFromAttributes(List<FlowFile> flowFiles, ProcessContext context, ProcessSession session, int batchSize) throws Exception {
+        Map<String, List<byte[]>> deletes = new HashMap<>();
+        for (FlowFile flowFile : flowFiles) {
+            String rowKey = context.getProperty(ROW_ID).evaluateAttributeExpressions(flowFile).getValue();
+            String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+            if (rowKey == null || rowKey.trim().equals("")) {
```

Comment:
why can't a key be made of spaces?

Reply:
Since this is a delete processor, you may be right. This would make more sense in a put because the scenarios where a user is likely to do all spaces by accident are more likely than intentional use.